trin-core/src/types/header.rs

```diff
@@ -194,6 +188,12 @@ fn try_value_into_u64(val: &Value) -> anyhow::Result<u64> {
 impl Decodable for Header {
     /// Attempt to decode a header from RLP bytes.
     fn decode(rlp: &Rlp) -> Result<Self, DecoderError> {
+        // decode raw big-endian encoded bytes for nonce
+        let mut nonce: [u8; 8] = Default::default();
+        let raw_nonce: Vec<u8> = rlp.val_at(14)?;
+        nonce.copy_from_slice(&raw_nonce[..]);
+        let nonce = u64::from_be_bytes(nonce);
```

Comment:
Ah, was this the problem with the genesis header decoding? That the actual RLP header definition expects the nonce to be treated as bytes, but the Rust implementation is treating it as a uint?

Maybe we should just change the Rust `Header` to treat the nonce as bytes internally?